src/main/java/de/tum/in/www1/artemis/service/metis/ConversationService.java

```diff
@@ -0,0 +1,165 @@
+            ConversationParticipantRepository conversationParticipantRepository, ConversationRepository conversationRepository, SimpMessageSendingOperations messagingTemplate) {
+        this.userRepository = userRepository;
+        this.courseRepository = courseRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.conversationRepository = conversationRepository;
+        this.messagingTemplate = messagingTemplate;
+    }
+
+    /**
+     * Persists given conversation
+     *
+     * @param courseId      id of course the conversation belongs to
+     * @param conversation  conversation to be persisted
+     * @return              persisted conversation
+     */
+    public Conversation createConversation(Long courseId, Conversation conversation) {
+        final User user = this.userRepository.getUserWithGroupsAndAuthorities();
+
+        if (conversation.getId() != null) {
+            throw new BadRequestAlertException("A new conversation cannot already have an ID", CONVERSATION_ENTITY_NAME, "idexists");
+        }
+
+        if (conversation.getConversationParticipants().isEmpty()
+                || conversation.getConversationParticipants().stream().anyMatch(conversationParticipant -> conversationParticipant.getUser().getId().equals(user.getId()))) {
+            throw new BadRequestAlertException("A new conversation must have other conversation participants", CONVERSATION_DETAILS_ENTITY_NAME, "invalidconversationparticipants");
+        }
+
+        final Course course = checkUserAndCourse(user, courseId);
+        conversation.setCourse(course);
+
+        Conversation savedConversation = conversationRepository.save(conversation);
+
+        ConversationParticipant conversationParticipantOfCurrentUser = new ConversationParticipant();
+        conversationParticipantOfCurrentUser.setUser(user);
+        conversation.getConversationParticipants().add(conversationParticipantOfCurrentUser);
+
+        conversation.getConversationParticipants().forEach(conversationParticipant -> conversationParticipantToCreate(conversationParticipant, savedConversation));
+        conversationParticipantRepository.saveAll(conversation.getConversationParticipants());
+        savedConversation.setConversationParticipants(conversation.getConversationParticipants());
+
+        // informs involved users about a new conversation
+        broadcastForConversation(new ConversationDTO(savedConversation, MetisCrudAction.CREATE));
+
+        return savedConversation;
+    }
+
+    /**
+     * fetch conversation from database by conversationId
+     *
+     * @param conversationId  id of the conversation to fetch
+     * @return  fetched conversation
+     */
+    public Conversation getConversationById(Long conversationId) {
```

Comment:
I personally think that having access to the repository over the service is a better approach for consistency.

Reply:
For the naming convention, it is also not done for every entity fetched and IMO it would be backward to follow this practice everywhere. For example, posts have many relations like answerposts, reactions, etc. In the codebase we do not specify the names by appending each of them which would be too long and unpractical.